helper-cli/src/main/kotlin/commands/SetDependencyRepresentationCommand.kt

```diff
@@ -0,0 +1,176 @@
+     */
+    private fun convertOrtResult(): Boolean =
+        convertResult<OrtResult> { result ->
+            println("Converting a full ORT result file.")
+
+            val analyzerResult = convertToTarget(result.analyzer?.result)
+            val convertedResult = result.copy(
+                analyzer = result.analyzer?.copy(result = analyzerResult)
+            )
+
+            writeResult(convertedResult)
+        }
+
+    /**
+     * Convert an [AnalyzerResult] file.
+     */
+    private fun convertAnalyzerResult(): Boolean =
+        convertResult<AnalyzerResult> {
+            println("Converting an analyzer result file.")
+
+            writeResult(convertToTarget(it))
+        }
+
+    /**
+     * Convert a [ProjectAnalyzerResult]. In this case, the result file contains an [AnalyzerResult], as a
+     * [ProjectAnalyzerResult] does not contain a dependency graph.
+     */
+    private fun convertProjectAnalyzerResult(): Boolean =
+        convertResult<ProjectAnalyzerResult> { result ->
+            println("Converting a project analyzer result file.")
+
+            val analyzerResult = AnalyzerResultBuilder().addResult(result).build()
+            writeResult(convertToTarget(analyzerResult))
+        }
+
+    /**
+     * Convert a result of type [T]. Read this result from the input file and invoke [block] with it. Return *false*
+     * if the result file could not be loaded (which probably means that it contains a result of a different type).
+     */
+    private inline fun <reified T : Any> convertResult(block: (T) -> Unit): Boolean =
+        runCatching {
+            block(ortFile.readValue())
+        }.fold(onSuccess = { true }, onFailure = { false })
+
+    /**
+     * Convert the given [analyzerResult] to the selected target format. Fail if no result is available or the
+     * result is already in the target format.
+     */
+    private fun convertToTarget(analyzerResult: AnalyzerResult?): AnalyzerResult {
+        val convertedResult = targetFormat.convert(
+            analyzerResult ?: failure("No analyzer result available.", 1)
+        )
+
+        if (convertedResult == analyzerResult) {
+            failure("The result already uses the dependency graph format. No conversion is done.", 2)
+        }
+
+        return convertedResult
+    }
+
+    /**
+     * Write the result of the conversion to the configured output file. If no output file was specified, overwrite the
+     * input file.
+     */
+    private inline fun <reified T : Any> writeResult(result: T): Boolean {
+        val targetFile = outputFile ?: ortFile
+
+        println("Writing converted result to $targetFile.")
+
+        targetFile.writeValue(result)
+        return true
+    }
+
+    /**
+     * Exit the program with the given failure [message] and status [code].
+     */
+    private fun failure(message: String, code: Int): Nothing {
+        println(message)
+        throw ProgramResult(code)
+    }
```

Comment:
I believe instead of this function we could simply throw a `UsageError` as all current causes are actually user errors.

Reply:
Done